File: Day261_2025/Q261_GFG.py
class Solution:
    def nextGreater(self, arr):
        n = len(arr)
        result = [-1] * n
        stack = []

        for i in range(2 * n):
            num = arr[i % n]
            while stack and arr[stack[-1]] < num:
                result[stack.pop()] = num
            if i < n:
                stack.append(i)
        
        return result

File: Day261_2025/Q261_GFG.cpp
class Solution {
  public:
    vector<int> nextGreater(vector<int> &arr) {
        // Q. Next Greater Element in Circular Array
        // code here
        int n = arr.size();
        vector<int> result(n, -1);
        stack<int> st;
        for (int i = 0; i < 2 * n; i++) {
            int num = arr[i % n];
            while (!st.empty() && arr[st.top()] < num) {
                result[st.top()] = num;
                st.pop();
            }
            if (i < n) {
                st.push(i);
            }
        }
        return result;
    }
};